src/aws_encryption_sdk/keyring/multi_keyring.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Resources required for Multi Keyrings."""
+import attr
+
+from aws_encryption_sdk.exceptions import EncryptKeyError
+from aws_encryption_sdk.keyring.base import Keyring
+
+try:  # Python 3.5.0 and 3.5.1 have incompatible typing modules
+    from typing import Iterable  # noqa pylint: disable=unused-import
+except ImportError:  # pragma: no cover
+    # We only actually need these imports when running the mypy checks
+    pass
+
+
+@attr.s
+class MultiKeyring(Keyring):
+    """Public class for Multi Keyring.
+
+    :param generator: Generator keyring used to generate data encryption key
+    :type generator: Keyring
+    :param list children: List of keyrings used to encrypt the data encryption key
+    :raises EncryptKeyError: if encryption of data key fails for any reason
+    """
+
+    generator = attr.ib(validator=attr.validators.instance_of(Keyring))
```

Comment:
Both generator and children need to be optional, with validation in `__attrs_post_init__` that at least one of them is set. You can see a similar pattern here: https://github.com/aws/aws-encryption-sdk-python/blob/keyring/src/aws_encryption_sdk/streaming_client.py#L99-L105